hypothesis/lms
# Canvas groups backend

# Testing notes

- Truncate your courses and grouping tables to test the insertion of a new course & groups
```truncate grouping, course;```

- Enable groups on the application instance

http://localhost:8001/admin/instance/Hypothesisb3be0b33d0b5e4b1cf3aaf04e0e1819a/

- Open an assignment with groups https://hypothesis.instructure.com/courses/125/assignments/873

- Check the grouping table, it should contain one row for the course and for each group (if opened as a teacher). Pay attention to the parent_id relationship, sections should point to the course.

```
-[ RECORD 1 ]-----------+-------------------------------------------------------------------------------------------------------
created                 | 2021-06-07 12:47:11.497763
updated                 | 2021-06-07 12:47:11.497763
id                      | 130
application_instance_id | 8
authority_provided_id   | 653e9620a656a954c684942f6443fa3e3410c03a
parent_id               | 
lms_id                  | f3cd019017839c4630358662a05540f2f6ec5f93
type                    | course
lms_name                | Developer Test Course with Sections Enabled
settings                | {"canvas": {"groups_enabled": true, "sections_enabled": true}, "blackboard": {"files_enabled": false}}
extra                   | {"canvas": {"custom_canvas_course_id": "125"}}
-[ RECORD 2 ]-----------+-------------------------------------------------------------------------------------------------------
created                 | 2021-06-07 12:47:11.619672
updated                 | 2021-06-07 12:47:11.619672
id                      | 131
application_instance_id | 8
authority_provided_id   | 381c455de84ae2f6d2fc84142f4e898b29d748f3
parent_id               | 130
lms_id                  | 132
type                    | canvas_group
lms_name                | another group
settings                | {}
extra                   | {"group_set_id": 121}
-[ RECORD 3 ]-----------+-------------------------------------------------------------------------------------------------------
created                 | 2021-06-07 12:47:11.619672
updated                 | 2021-06-07 12:47:11.619672
id                      | 132
application_instance_id | 8
authority_provided_id   | 102a25279bc609179f58cb4bf5c50d7dfc6f0929
parent_id               | 130
lms_id                  | 130
type                    | canvas_group
lms_name                | others
settings                | {}
extra                   | {"group_set_id": 121}
-[ RECORD 4 ]-----------+-------------------------------------------------------------------------------------------------------
created                 | 2021-06-07 12:47:11.619672
updated                 | 2021-06-07 12:47:11.619672
id                      | 133
application_instance_id | 8
authority_provided_id   | 802107eead62d51dcdef0f317f0eb0b9868eaa7c
parent_id               | 130
lms_id                  | 129
type                    | canvas_group
lms_name                | Seans
settings                | {}
extra                   | {"group_set_id": 121}
-[ RECORD 5 ]-----------+-------------------------------------------------------------------------------------------------------
created                 | 2021-06-07 12:47:11.619672
updated                 | 2021-06-07 12:47:11.619672
id                      | 134
application_instance_id | 8
authority_provided_id   | e8c66a347a5f547ae87928d4e92e5045908f8d54
parent_id               | 130
lms_id                  | 131
type                    | canvas_group
lms_name                | test
settings                | {}
extra                   | {"group_set_id": 121}
```

- The client should show all the groups in the top left menu.

- Open speed grader,  it should open the client for students that launched the assignment.

- Open the same assignment as a student. The menu should show only one group (no selection possible).
